Tests/MongoSwiftTests/ReadWriteConcernTests.swift

```diff
@@ -0,0 +1,213 @@
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+import libmongoc
+
+final class ReadWriteConcernTests: XCTestCase {
+	static var allTests: [(String, (ReadWriteConcernTests) -> () throws -> Void)] {
+        return [
+        	("testReadConcernType", testReadConcernType),
+            ("testClientReadConcern", testClientReadConcern),
+            ("testDatabaseReadConcern", testDatabaseReadConcern),
+            ("testOperationReadConcerns", testOperationReadConcerns)
+        ]
+    }
+
+    override func setUp() {
+    	self.continueAfterFailure = false
+    }
+
+    func testReadConcernType() throws {
+    	// check that setting readConcern level to all valid inputs doesn't throw
+    	expect(try ReadConcern(.local)).toNot(throwError())
+    	expect(try ReadConcern(.majority)).toNot(throwError())
+    	expect(try ReadConcern(.available)).toNot(throwError())
+    	expect(try ReadConcern(.linearizable)).toNot(throwError())
+    	expect(try ReadConcern(.snapshot)).toNot(throwError())
+
+    	expect(try ReadConcern("local")).toNot(throwError())
+    	expect(try ReadConcern("majority")).toNot(throwError())
+    	expect(try ReadConcern("available")).toNot(throwError())
+    	expect(try ReadConcern("linearizable")).toNot(throwError())
+    	expect(try ReadConcern("snapshot")).toNot(throwError())
+
+    	// check level var works as expected
+    	let rc = try ReadConcern(.majority)
+    	expect(rc.level).to(equal("majority"))
+
+    }
+
+    func testClientReadConcern() throws {
+    	// create a client with no options and check its RC
+    	let client1 = try MongoClient()
+    	// expect the readConcern property to exist with a nil level
+    	expect(client1.readConcern.level).to(beNil())
+
+    	// expect that a DB created from this client inherits its unset RC 
+    	let db1 = try client1.db("test")
+    	expect(db1.readConcern.level).to(beNil())
+
+    	// expect that a DB created from this client can override the client's unset RC
+    	let db2 = try client1.db("test", options: DatabaseOptions(readConcern: try ReadConcern(.majority)))
+    	expect(db2.readConcern.level).to(equal("majority"))
+
+    	client1.close()
```

Comment:
do these need to be called explicitly? If you're concerned about memory the deinitializer will be called later